loopin.go

```diff
@@ -445,6 +463,17 @@ func (s *loopInSwap) waitForHtlcConf(globalCtx context.Context) (
 		}
 	}
 
+	// Store htlc tx hash for accounting purposes. Usually this call is a
+	// no-op because the htlc tx hash was already known. Exceptions are:
+	//
+	// - Old pending swaps that were initiated before we persisted the htlc
+	//   tx hash directly after publish.
+	//
+	// - Swaps that experienced a crash during their call to SendOutputs. In
+	//   that case, we weren't able to record the tx hash.
+	txHash := conf.Tx.TxHash()
+	s.htlcTxHash = &txHash
```

Comment:
Is it worth saving state here if our hash previously was not stored?
Thinking about one of these edge cases where our htlc confirms, then we restart and the txid is lost. We then run through this code again, and might pick up another htlc (which we could avoid if we'd saved here). 

This is probably the edge of edge cases though.

Reply:
I thought about that, but we are indeed already in an unsafe situation with us not having persisted the htlc tx hash in the first place. This would also only be useful if the conf notifications are returned non-deterministically, which doesn't seem to be the case. I think it doesn't outweigh having a fingerprint of this in the database.